### PR TITLE
ARTEMIS-3024 Expiry or DLQ Retry not working with AMQP

### PR DESCRIPTION
**Pull request by clebertsuconic**

### CODE REVIEW

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java [hunk 1]
```diff
@@ -2140,6 +2140,7 @@ private int iterQueue(final int flushLimit,
                                       QueueIterateAction messageAction) throws Exception {
       int count = 0;
       int txCount = 0;
+      Integer expectedHits = messageAction.expectedHits();
```
Comment (michaelandrepearce) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java:2143]: Whats the purpose of this?

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/server/Queue.java [hunk 2]
```diff
@@ -361,6 +361,10 @@ int moveReferences(int flushLimit,
 
    int retryMessages(Filter filter) throws Exception;
 
+   default int retryMessages(Filter filter, Integer expectedHits) throws Exception {
```
Comment (michaelandrepearce) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/server/Queue.java:364]: Why default method? Its been implemented in queueimpl

Comment (clebertsuconic) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/server/Queue.java:364]: there are a bunch of Fakes that don't require implementing anything.

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java (hunk 1 above)
Comment (clebertsuconic) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java:2143]: QueueControl.expire with a messageID will only expire a single record, however the iteration will go until the end. it's sort of an optimization.

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/server/Queue.java (hunk 2 above)
Comment (clebertsuconic) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/server/Queue.java:364]: @michaelandrepearce I could go without the default.. but having the default here minimized the number of changes on Fakes.. etc... It was just for simplicity and minimize the number of lines changed.

**Pull request merged**